smack-experimental/src/main/java/org/jivesoftware/smackx/chat_markers/ChatMarkersManager.java

```diff
@@ -218,4 +224,41 @@ public synchronized boolean removeIncomingChatMarkerMessageListener(ChatMarkersL
         }
         return res;
     }
+
+    /**
+     * Send a message stanza to the recipient defined in the <tt>To</tt> getter from the <tt>Message</tt>.
+     *
+     * @param message          instance of {@link Message} with a To previous defined.
+     * @param chatMarkersState one of the values given in {@link ChatMarkersState}.
+     * @param messageId        id of the message to be updated.
+     * @throws NotConnectedException if the connection is not connected.
+     * @throws InterruptedException  if the connection is interrupted.
+     * @throws IllegalStateException if one of the params don't match the rules.
+     */
+    public void markMessage(Message message, ChatMarkersState chatMarkersState, String messageId)
+            throws
+            NotConnectedException,
+            InterruptedException,
+            IllegalStateException {
+        if (message == null) {
+            throw new IllegalStateException("Message must not be null");
+        }
+
+        if (message.getTo() == null) {
+            throw new IllegalStateException("To attribute must not be null");
+        }
+
+        if (message.getStanzaId() == null) {
+            message.setStanzaId();
+        }
+
+        if (chatMarkersState == ChatMarkersState.received) {
```

Comment:
This might be a good place for a switch-case environment?

Reply:
switched